modules/bm_sim/include/bm_sim/P4Objects.h

```diff
@@ -61,7 +61,9 @@ class P4Objects {
   explicit P4Objects(std::ostream &outstream = std::cout)
       : outstream(outstream) { }
 
-  int init_objects(std::istream *is, int device_id = 0, size_t cxt_id = 0,
+  int init_objects(std::istream *is,
+                   LookupStructureFactory * lookup_factory,
```

Comment:
I'd rather have the lookup_factory be a static member of `MatchUnitGeneric`. The use case I see is that each bmv2 target will specify its own factory once (or use the default one) and passing this pointer around is just making things more complicated. Do you see any drawback with that?